### PR TITLE
feat(tests): Add mobile screenshot tests

### PR DESCRIPTION
**Pull request by priscilawebdev**
closes: INGEST-314

Added tests for the mobile screenshot feature. Related PRs:

https://github.com/getsentry/sentry/pull/28309
https://github.com/getsentry/sentry/pull/27350
https://github.com/getsentry/sentry/pull/27274
https://github.com/getsentry/sentry/pull/27228

### CODE REVIEW

File: tests/js/sentry-test/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import {BoundFunctions, FindByText} from 'sentry-test/reactTestingLibrary';
+
+/**
+ * Search for a text broken up by multiple html elements
+ * e.g.: <div>Hello <span>world</span></div>
+ */
+export async function findByTextContent(
```
Comment (priscilawebdev) [inline tests/js/sentry-test/utils.tsx:7]: I think in the future we will have more utilities in this file

File: tests/js/spec/components/events/interfaces/stacktrace.spec.tsx [hunk 2]
```diff
@@ -27,10 +27,6 @@ function renderedComponent(
 }
 
 describe('StackTrace', function () {
```
Comment (priscilawebdev) [inline tests/js/spec/components/events/interfaces/stacktrace.spec.tsx:29]: this should not be part of this PR, but I could not resist, pls lmk if that is fine 😄

File: static/app/components/events/eventTagsAndScreenshot/screenshot/index.tsx [hunk 3]
```diff
@@ -118,22 +115,20 @@ function Screenshot({event, attachments, organization, projectSlug, onDelete}: P
   }
 
   return (
-    <Role role={organization.attachmentsRole}>
+    <Role organization={organization} role={organization.attachmentsRole}>
```
Comment (priscilawebdev) [inline static/app/components/events/eventTagsAndScreenshot/screenshot/index.tsx:118]: the `Role` component is already wrapped with `withOrganization` but passing the org here was needed for the test to work. It was the simplest solution

File: static/app/components/events/eventTagsAndScreenshot/index.tsx [hunk 4]
```diff
@@ -8,49 +8,56 @@ import Tags from './tags';
...
+  organization,
+  isShare = false,
+  isBorderless = false,
+  hasContext = false,
 }: Props) {
   const {tags = []} = event;
 
-  if (!tags.length && !hasContext && isShare) {
+  const screenshot = attachments.find(
+    ({name}) => name === 'screenshot.jpg' || name === 'screenshot.png'
```
Comment (priscilawebdev) [inline static/app/components/events/eventTagsAndScreenshot/index.tsx:34]: this new condition is described in the draft PR: https://github.com/getsentry/develop/pull/422

File: tests/js/sentry-test/utils.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  screen: BoundFunctions<{findByText: FindByText}>,
+  textMatch: string | RegExp
+): Promise<HTMLElement> {
+  return screen.findByText((_, contentNode) => {
+    const hasText = (node: Element) => node.textContent === textMatch;
```
Comment (markstory) [inline tests/js/sentry-test/utils.tsx:12]: Would you be able to use `node.innerText` to search the element and the text of its children?

Comment (priscilawebdev) [reply tests/js/sentry-test/utils.tsx:12]: no it has to be `textContent`, otherwise it will always be `undefined`

Comment (priscilawebdev) [reply tests/js/sentry-test/utils.tsx:12]: It seems that `innerText` is not implemented in `jsdom`.. see [here](https://github.com/jsdom/jsdom/issues/1245).

**Review by matejminar — APPROVED** [review]
